app/models/user.rb

```diff
@@ -15,4 +18,16 @@ def normalize_queue_item_list
   def queued_video?(video)
     queue_items.find_by(video: video)
   end
+
+  def already_followed?(leader)
+    following_relationships.map(&:leader).include?(leader)
+  end
+
+  def is_the?(role)
+    self == role
+  end
+
+  def not_followable?(user)
+    self.already_followed?(user) || self.is_the?(user)

```

Comment:
Its hard to follow this - `self.is_the?(user)` first, you pass in user, but when I look at the `is_the?` method user changes to role.